### PR TITLE
Enabled SHARP test scenarios

### PR DESCRIPTION
**Pull request by itemko**

### CODE REVIEW

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 12-Sep-2019

Artefact file | Status
-- | --
[build_nccl.log](https://gist.githubusercontent.com/mellanox-github/14e16e9bf34c01c06ba89062f47232b2/raw/7c51e087d8eeec4e6c86b5269f16f933e02ce079/build_nccl.log) | :heavy_check_mark: PASS
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/14e16e9bf34c01c06ba89062f47232b2/raw/92f64db96d5fd857f8e5f9d8170e9423ed226b87/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/14e16e9bf34c01c06ba89062f47232b2/raw/83cb1902e9f7e5af80b395664bec3ada7bfb9cfc/configure_sharp_startup.log) | :heavy_check_mark: PASS
[configure_sharp_stop.log](https://gist.githubusercontent.com/mellanox-github/14e16e9bf34c01c06ba89062f47232b2/raw/1eaa64a4a5a6799f3e0b7a59336815edd6e8afc9/configure_sharp_stop.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/14e16e9bf34c01c06ba89062f47232b2/raw/200ed38758a015787e265e876413e16ad03d2a74/run_nccl_test.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 12-Sep-2019

Artefact file | Status
-- | --
[build_nccl.log](https://gist.githubusercontent.com/mellanox-github/cd148a6052454f24b320654f8106cb9f/raw/e949506a266430dea2a196572f5080c779f112b0/build_nccl.log) | :heavy_check_mark: PASS
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/cd148a6052454f24b320654f8106cb9f/raw/357d29e94ef2165311fe4c4c1251d12ee15f9038/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/cd148a6052454f24b320654f8106cb9f/raw/74f70d5e745a6a2705d47fd0378db03b4e7fc671/configure_sharp_startup.log) | :heavy_check_mark: PASS
[configure_sharp_stop.log](https://gist.githubusercontent.com/mellanox-github/cd148a6052454f24b320654f8106cb9f/raw/3b086ab7bea71d2902ba648b6b0ace52e3ec305f/configure_sharp_stop.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/cd148a6052454f24b320654f8106cb9f/raw/b6a01ead555c5a2e4b061686863393d403103295/run_nccl_test.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

File: .ci/configure_sharp.sh [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+    fi
+
+    echo "INFO: check_opensm_conf on ${SHARP_AM_NODE}... DONE"
+}
+
+verify_sharp() {
+    echo "INFO: verify_sharp..."
+
+    export PATH="${SHARP_DIR}/bin:$PATH"
+    export LD_LIBRARY_PATH="${SHARP_DIR}/lib:${NCCL_DIR}/lib:${LD_LIBRARY_PATH}"
```
Comment (mike-dubman) [inline .ci/configure_sharp.sh]: it should be set by ```module load hpcx-gcc```
maybe nccl vars should be placed into /path/to/nccl/etc/nccl.module and load it?

well done PR!

Comment (itemko) [reply .ci/configure_sharp.sh]: Fixed for SHARP_DIR - now we fully rely on `module load hpcx-gcc` and HPCX_SHARP_DIR.
For NCCL_DIR I'm planning to implement nightly NCCL testing and NCCL env may be set after that via something like `module load dev/nccl-nightly`.

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI status: <b>PASS</b#  > (click for details)</summary>

**Note:** the artefacts will be deleted after 16-Sep-2019

Artefact file | Status
-- | --
[build_nccl.log](https://gist.githubusercontent.com/mellanox-github/f9847ffb6bb2225f89674b1dd5db7477/raw/b82cb605c81f11fb2543591df99bdf7c0e3a0b3e/build_nccl.log) | :heavy_check_mark: PASS
[build_nccl_rdma_sharp_plugins.log](https://gist.githubusercontent.com/mellanox-github/f9847ffb6bb2225f89674b1dd5db7477/raw/d984acf70d2827331ba7efe313bad44c547ad016/build_nccl_rdma_sharp_plugins.log) | :heavy_check_mark: PASS
[configure_sharp_startup.log](https://gist.githubusercontent.com/mellanox-github/f9847ffb6bb2225f89674b1dd5db7477/raw/a268d54d0c1fbc5e043631ddfeacdfeb221a8b96/configure_sharp_startup.log) | :heavy_check_mark: PASS
[configure_sharp_stop.log](https://gist.githubusercontent.com/mellanox-github/f9847ffb6bb2225f89674b1dd5db7477/raw/c67160bcfa09c9d55c1c3fa11786f50e297c497d/configure_sharp_stop.log) | :heavy_check_mark: PASS
[run_nccl_test.log](https://gist.githubusercontent.com/mellanox-github/f9847ffb6bb2225f89674b1dd5db7477/raw/0807e40d5d146b84b1be7b7a90f0521c4f4dc5c4/run_nccl_test.log) | :heavy_check_mark: PASS

**Comment by mellanox-github** [comment]
Mellanox CI PASSed.

**Comment by mike-dubman** [comment]
+1

**Comment by itemko** [comment]
@bureddy 
Could you please merge this PR?

**Pull request merged**